synapse/common.py

```diff
@@ -1390,6 +1368,31 @@ def _timeout(delay):
     """
     loop = asyncio.get_running_loop()
     return _Timeout(loop.time() + delay if delay is not None else None)
+# End - Vendored Code from Python 3.12+
+
+async def waitretn(futu, timeout):
```

Comment:
why did this need to be moved?

Reply:
@vEpiphyte suggested it should be moved under the vendored code.